cmd/action.go

```diff
@@ -0,0 +1,265 @@
+	var filePath, header string
+
+	cmd := &cli.Command{
+		Use:   "create",
+		Short: "Create an action",
+		Args:  cli.NoArgs,
+		Example: heredoc.Doc(`
+			$ shield action create --file=<action-body> --header=<key>:<value>
+		`),
+		Annotations: map[string]string{
+			"action:core": "true",
+		},
+		RunE: func(cmd *cli.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+
+			var reqBody shieldv1beta1.ActionRequestBody
+			if err := parseFile(filePath, &reqBody); err != nil {
+				return err
+			}
+
+			err := reqBody.ValidateAll()
+			if err != nil {
+				return err
+			}
+
+			host := appConfig.App.Host + ":" + strconv.Itoa(appConfig.App.Port)
+			ctx := context.Background()
+			client, cancel, err := createClient(ctx, host)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			ctx = setCtxHeader(ctx, header)
+
+			res, err := client.CreateAction(ctx, &shieldv1beta1.CreateActionRequest{
+				Body: &reqBody,
+			})
+			if err != nil {
+				return err
+			}
+
+			spinner.Stop()
+			logger.Info(fmt.Sprintf("successfully created action %s with id %s", res.GetAction().GetName(), res.GetAction().GetId()))
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&filePath, "file", "f", "", "Path to the action body file")
+	cmd.MarkFlagRequired("file")
+	cmd.Flags().StringVarP(&header, "header", "H", "", "Header <key>:<value>")
+	cmd.MarkFlagRequired("header")
+
+	return cmd
+}
+
+func updateActionCommand(logger log.Logger, appConfig *config.Shield) *cli.Command {
+	var filePath string
+
+	cmd := &cli.Command{
+		Use:   "update",
+		Short: "Edit an action",
+		Args:  cli.ExactArgs(1),
+		Example: heredoc.Doc(`
+			$ shield action update <action-id> --file=<action-body>
+		`),
+		Annotations: map[string]string{
+			"action:core": "true",
+		},
+		RunE: func(cmd *cli.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+
+			var reqBody shieldv1beta1.ActionRequestBody
+			if err := parseFile(filePath, &reqBody); err != nil {
+				return err
+			}
+
+			err := reqBody.ValidateAll()
+			if err != nil {
+				return err
+			}
+
+			host := appConfig.App.Host + ":" + strconv.Itoa(appConfig.App.Port)
+			ctx := context.Background()
+			client, cancel, err := createClient(ctx, host)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			actionID := args[0]
+			_, err = client.UpdateAction(ctx, &shieldv1beta1.UpdateActionRequest{
+				Id:   actionID,
+				Body: &reqBody,
+			})
+			if err != nil {
+				return err
+			}
+
+			spinner.Stop()
+			logger.Info(fmt.Sprintf("successfully updated action with id %s", actionID))
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&filePath, "file", "f", "", "Path to the action body file")
+	cmd.MarkFlagRequired("file")
+
+	return cmd
+}
+
+func getActionCommand(logger log.Logger, appConfig *config.Shield) *cli.Command {
+	cmd := &cli.Command{
+		Use:   "get",
```

Comment:
command name should also be renamed to `view` instead of get.

Reply:
oh, okay my bad